### DISCUSSION TITLE
Annotated Dependency functions/Callables Aren't Executed on Version 0.94.1

### DISCUSSION THREAD

**Discussion by grdworkin** [fastapi/fastapi#9297] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import Depends, FastAPI

app = FastAPI()

async def _add_1(i: str):
    return int(i) + 1

@app.get("/add_1/")
async def add_1(i: Annotated[int, Depends(_add_1)]):
    return i

class _Add_2:
    def __call__(self, i: str):
        return int(i) + 2

_add_2 = _Add_2()

@app.get("/add_2/")
async def add_2(i: Annotated[int, Depends(_add_2)]):
    return i
```

### Description

/add_1/?i=1 should return 2 but returns 1 and /add_2/?i=1 should return 3 but also returns 1.

Related: https://github.com/tiangolo/fastapi/discussions/9277

Updating to 0.95.0 fixes this problem.

I spent a while debugging this and am hoping this helps out others. I apologize if this isn't the right way of going about this.

The new annotated dependencies feature is great but it's kind of unfortunate that it doesn't error out on old code and produces silent failures. I propose that the documentation is updated to call out how new the feature is or that FastAPI needs to be at least version 0.95.0 specifically because older versions "work" with annotated. I'll look into putting forward a pull request for it.

### Operating System

macOS

### FastAPI Version

0.94.1

### Python Version

3.11

**Comment by Kludex** [comment, marked as answer]
Yeah, the feature was introduced on FastAPI 0.95.0.
